src/components/player/player.js

```diff
@@ -1,11 +1,77 @@
-import React from "react"
+import React, { useEffect, useState } from "react"
 import { PlayButton } from "react-soundplayer/components"
 import { withCustomAudio } from "react-soundplayer/addons"
 
+import fetchJsonp from "fetch-jsonp"
+
 const Player = withCustomAudio(props => {
+  const [streamData, setStreamData] = useState({ streamstatus: 1 })
+  const [mixcloudData, setMixcloudData] = useState("")
+
+  const fetchMixcloudStream = url => {
+    if (!url) url = "https://api.mixcloud.com/entropyfm/feed/"
+    fetch(url)
+      .then(resp => resp.json())
+      .then(data => {
+        for (let i = 0; i < 10; i++) {
+          // 10 tries to fetch a random mixcloud recording
+          let temp = data.data[Math.floor(Math.random() * data.data.length)]
+          if (temp.cloudcasts) {
+            setMixcloudData(temp.cloudcasts[0].key)
+            return
+          }
+        }
+        // No mixcloud recording found, go to next page
+        if (data.paging.next) fetchMixcloudStream(data.paging.next)
+      })
+  }
+
+  const fetchMetadata = () => {
+    fetchJsonp("https://s28.myradiostream.com/26952/statistics?sid=1&json=1")
+      .then(resp => resp.json())
+      .then(data => {
+        setStreamData(data.streams[0])
+
+        // Check if stream is down, if so fetch mixcloud livestream
+        if (!streamData.streamstatus && !mixcloudData) fetchMixcloudStream()
+      })
+      .catch(() => {
+        // In case we're unable to get the json data
+        if (streamData.streamstatus) setStreamData({ streamstatus: 0 })
+        if (!mixcloudData) fetchMixcloudStream()
+      })
+  }
+
+  useEffect(() => {
+    fetchMetadata()
+    const timer = setInterval(() => fetchMetadata(), 10000)
+
+    return () => clearInterval(timer)
+  })
+
   return (
     <div className="player">
-      <PlayButton className="button" {...props} />
+      {streamData.streamstatus ? (
+        <div>
+          <PlayButton
+            className="button"
+            {...props}
+            style={{ float: "left", marginRight: "10px" }}
+          />
+          <p>{streamData.songtitle}</p>
+        </div>
+      ) : (
+        <p>We're currently offline, check out a previous show below!</p>
```

Comment:
This also belongs in siteMetadata. We should probably break that out into separate queries at some point, but for now I'm fine bundling it all there.